### PR TITLE
Proxy update

### PR DESCRIPTION
**Pull request by pyrodie18**
##### SUMMARY
Initial changes to server role for 2.0.0

**NOTE**:  This needs to merge after 780

##### COMPONENT NAME
Proxy Role

##### ADDITIONAL INFORMATION
Major Changes include:

- Updating Molecule matrix to test all OS, with all versions, and all databases (#777 )
- Updated documentation
- Removed support for anything < 5.0 (#766)
- Cleaned up handling of variables between OSs
- Added Ubuntu 22.04 (#763)
- Added Support for Debian 11 (#764)
- Removed support for EOL operating systems
- Cleaned up some variables (#768 )

### CODE REVIEW

**Comment by pyrodie18** [comment]
Need to rebase

File: changelogs/fragments/proxy.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+breaking_changes:
+  - zabbix_server_host - Removed zabbix_server_host and replaced with zabbix_proxy_server
```
Comment (D3DeFi) [inline changelogs/fragments/proxy.yml:2]: This format assumes that each chnagelog fragment will be assigned correct reference to module or role. Unfortunately, this is not the case and the resulting changlog format will be hard to read. 

That is why we prepend each line with 
```
  - zabbix_proxy (role) - Removed zabbix_server_host and
```

File: docs/ZABBIX_PROXY_ROLE.md [hunk 2]
```diff
@@ -53,41 +55,33 @@ ansible-galaxy collection install ansible.posix
 
 ### MySQL
 
-When you are a MySQL user and using Ansible 2.10 or newer, then there is a dependency on the collection named `community.mysql`. This collections are needed as the `mysql_` modules are now part of collections and not standard in Ansible anymmore. Installing the collection:
+When you are a MySQL there is a dependency on the collection named `community.mysql`.  Installing the collection:
```
Comment (D3DeFi) [inline docs/ZABBIX_PROXY_ROLE.md:58]:
```suggestion
When you are using MySQL, there is a dependency on the collection named `community.mysql`.  Installing the collection:
```

File: docs/ZABBIX_PROXY_ROLE.md [hunk 3]
```diff
@@ -53,41 +55,33 @@ ansible-galaxy collection install ansible.posix
...
 
 ```sh
 ansible-galaxy collection install community.mysql
 ```
 
 ### PostgreSQL
 
-When you are a PostgreSQL user and using Ansible 2.10 or newer, then there is a dependency on the collection named `community.postgresql`. This collections are needed as the `postgresql_` modules are now part of collections and not standard in Ansible anymmore. Installing the collection:
+When you are a PostgreSQL there is a dependency on the collection named `community.postgresql`. Installing the collection:
```
Comment (D3DeFi) [inline docs/ZABBIX_PROXY_ROLE.md:66]:
```suggestion
When you are using PostgreSQL,
there is a dependency on the collection named `community.postgresql`. Installing the collection:
```

Similar typos found later as well

File: docs/ZABBIX_PROXY_ROLE.md [hunk 4]
```diff
@@ -224,7 +216,7 @@ We need to have the following dependencies met:
 
 ```yaml
 zabbix_proxy_database: mysql
-zabbix_proxy_database_long: mysql
+zabbix_db_type_long: mysql
```
Comment (D3DeFi) [inline docs/ZABBIX_PROXY_ROLE.md:219]: I may be mistaken, but does this deviate official ansible best practices where role defaults should be prefixed with role name? 

Not a suggestion, just an invitation for a discussion.

**Review by D3DeFi — APPROVED** [review]
I ignored zabbix_server changes, those I guess need rebase as you said. Proxy changes look good. I left some recommendations/tips in the review, but those are no  stops for the merge.

**Comment by D3DeFi** [comment]
And also, amazing job with all of this ❤️

File: docs/ZABBIX_PROXY_ROLE.md (hunk 4 above)
Comment (pyrodie18) [reply docs/ZABBIX_PROXY_ROLE.md:219]: Makes sense.

**Comment by pyrodie18** [comment]
Closing this and will resubmit shortly

**Pull request closed without merging**